src/sqswatcher/sqswatcher.py

```diff
@@ -220,51 +238,104 @@ def _retrieve_all_sqs_messages(queue, max_processed_messages):
             # looping until receive_messages returns at least 1 message
             break
 
-    log.info("Retrieved %d messages from SQS queue", len(messages))
+    log.info("Retrieved {} messages from SQS queue {}".format(len(messages), queue_name))
 
     return messages
 
 
-def _parse_sqs_messages(sqs_config_region, sqs_config_proxy, messages, table, queue):
-    add_events = []
-    remove_events = []
-    for message in messages:
-        message_text = json.loads(message.body)
-        message_attrs = json.loads(message_text.get("Message"))
-
-        event_type = message_attrs.get("Event")
-        if not event_type:
-            log.warning("Unable to read message. Deleting.")
-            message.delete()
-            continue
-
-        if event_type == "parallelcluster:COMPUTE_READY":
-            add_event = _process_compute_ready_event(sqs_config_region, sqs_config_proxy, message_attrs, message, table)
-            if add_event:
-                add_events.append(add_event)
-        elif event_type == "autoscaling:EC2_INSTANCE_TERMINATE":
-            remove_event = _process_instance_terminate_event(message_attrs, message, table, queue)
-            if remove_event:
-                remove_events.append(remove_event)
-        else:
-            log.info("Unsupported event type %s. Discarding message." % event_type)
-            message.delete()
-
+def _parse_sqs_messages(sqs_config, messages, table, queue):
     update_events = OrderedDict()
-    for event in itertools.chain(add_events, remove_events):
-        log.info("Processing %s event for instance %s", event.action, event.host)
+    parsed_events = _parse_messages_helper(messages, False, sqs_config, table, queue)
+    for event in itertools.chain(parsed_events["ADD"], parsed_events["REMOVE"]):
+        log.info("Processing {} event for instance {}".format(event.action, event.host))
         hostname = event.host.hostname
         if hostname in update_events:
             # events are looped in the order they are fetched and by iterating over REMOVE events after ADD events.
             # in case of collisions let's always remove the item that comes first in order to always favor REMOVE ops.
-            log.info("Hostname collision. Discarding event %s in favour of event %s", update_events[hostname], event)
+            log.info(
+                "Hostname collision. Discarding event {} in favour of event {}".format(update_events[hostname], event)
+            )
             update_events[hostname].message.delete()
             del update_events[hostname]
         update_events[hostname] = event
 
     return update_events.values()
 
 
+def _parse_health_messages(sqs_config, messages, health_queue):
+    health_update_events = OrderedDict()
+    instance_id_to_hostname = get_cluster_instance_info(sqs_config.stack_name, sqs_config.region, include_master=False)
```

Comment:
I would have personally extracted the instance-ids from all health messages and then performed a single (or a few if needed) ec2 calls by passing an array of instance-ids. 
I expect health messages to be always a few (more like 1 at a time) and, in case of big clusters, describing all instances rather than 1 seems a bit overkill. What do you think?

of if it gets to complicated to parse the messages twice you could also perform a describe instance for each message in the queue since we don't expect them to be many